support-frontend/test/services/stepfunctions/SupportWorkersClientTest.scala

```diff
@@ -60,4 +60,28 @@ class SupportWorkersClientTest extends AnyFlatSpec with Matchers with MockitoSug
     actual shouldBe StatusResults.failure(CheckoutFailureReasons.Unknown)
   }
 
+  "generateExecutionName" should "generate a valid name" in {
+    def doTest(len: Int, value: Long, truncated: Boolean) = {
+      import Client.generateExecutionName
+      val actual = generateExecutionName(("123456789-" * 10).take(len), value)
+      info(s"string: <$actual>")
+      withClue(s"string: <$actual>") {
+        actual.length should be(80)
+        actual should fullyMatch regex ("[0-9a-zA-Z_-]+")
+        if (truncated)
+          actual.takeRight(2) should be("--")
+        else
+          actual.last should not be (('-'))
+      }
+    }
+    doTest(68, Long.MinValue, truncated = false)
+    doTest(69, Long.MinValue, truncated = true)
+    doTest(68, Long.MaxValue, truncated = false)
+    doTest(69, Long.MaxValue, truncated = true)
+    doTest(68, 0L, truncated = false)
+    doTest(69, 0L, truncated = true)
+    doTest(68, System.nanoTime(), truncated = false)
+    doTest(69, System.nanoTime(), truncated = true)
```

Comment:
I quite like [table driven testing](https://www.scalatest.org/user_guide/table_driven_property_checks) for this type of test - just sharing as this is nice and clear as it is.

Reply:
thanks for the pointer, that's a nice level of separation.

On first impression only thing I don't like is it separates the data from its context - as it is, the parameter names are all there, cmd-click works well, and the stack trace will nicely show you where it was working at the time.

But I am aiming to learn more about the scalatest functionality so I will play with that next time and see how it feels.